### PR TITLE
Created simple sensor smoothing library

### LINKED ISSUE #23: create smoothing library
Used to get an average of recent sensor data.

### PR DESCRIPTION
**Pull request by nBukaty**
Calculates moving average of a user defined value for easier sensor reading. The issues I was having in lab should be fixed now. Code has been tested and works but I still need to test on the actual board.
Closes #23

### CODE REVIEW

File: arduino/sensors/sensor_smoothing/smoothing.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	readings[readIndex] = newValue;
+	total = total + readings[readIndex];
+	readIndex = readIndex + 1;
+
+	if (readIndex >= numReadings)
+	{
+		readIndex = 0;
+	}
+
+	newAvg = total/numReadings;
```
Comment (Amertz08) [inline arduino/sensors/sensor_smoothing/smoothing.cpp]: Fix spacing

Comment (nBukaty) [reply arduino/sensors/sensor_smoothing/smoothing.cpp]: In what way? After the bracket should be there if that's what you mean.

Comment (Amertz08) [reply arduino/sensors/sensor_smoothing/smoothing.cpp]: Space between the `/` operator. `total / numReadings` not `total/numReadings`

**Comment by Amertz08** [comment]
I think you should consider using `this` pointer for member variables and methods as it is more explicit.

**Comment by Amertz08** [comment]
Makes it easier to differentiate between members and non member functions/variables

**Comment by nBukaty** [comment]
I think that adds a lot of clutter and is unnecessary. I'd prefer using "_" or something like that if you want to make things more clear.

**Comment by Amertz08** [comment]
`_` is fine for private parameters and private methods

**Comment by Amertz08** [comment]
And is actually imo preferential

**Comment by nBukaty** [comment]
@cduddy if I'm not mistaken most sensors will read out a whole number. The floats were just to stay consistent with the already converted cm value.

**Pull request merged**